tests/commands/test_rename.py

```diff
@@ -20,3 +19,37 @@ def test_rename_run(tmp_library: TemporaryLibrary) -> None:
     doc, = db.query_dict({"title": old_title})
     assert doc.get_main_folder_name() == new_name
     assert os.path.exists(doc.get_main_folder())
+
+
+def test_regenerate_name(tmp_library: TemporaryLibrary,
+                         tmp_config: TemporaryConfiguration,
+                         ) -> None:
```

Comment:
Not sure what's wrong with the tests, but from a quick skim, this should just have a `TemporaryLibrary` (that inherits from `TemporaryConfiguration`). Maybe having two confuses the tests?

Reply:
Could be! We'll se on the next run :thinking:

Reply:
So I am looping over the picked documents. That list is shuffling after each run :open_mouth:

Reply:
There is something strange when running under Pytest and I'm unable to figure out what makes the list mutate after each run. It was as if the document list was being sorted within each iteration. But, as the level of abstraction is now so high, I just can't understand why.